### PR TITLE
[MOBILE-1567] Named user attributes

### PR DESCRIPTION
**Pull request by oristanovic**
### What do these changes do?
Add names user attributes support.

### Why are these changes necessary?
Ti allow customers using React Native to use this feature.

### How did you verify these changes?
Manual tests in sample apps.

### CODE REVIEW

File: src/js/UrbanAirship.js [hunk 1]
```diff
@@ -324,6 +324,29 @@ class UrbanAirship {
     });
   }
 
+  /**
+   * Creates an editor to modify the named user attributes.
+   *
+   * @return {Promise.<AttributeEditor, Error>}  A promise that returns an attribute editor instance if resolved, or an Error if the
```
Comment (BrianBatchelder) [inline src/js/UrbanAirship.js:330]: The SDKs allow attributes to be set before setting a named user id. The attributes are uploaded once a named user id has been set. Why did you decide to force the named user id to be set before setting attributes?

Comment (oristanovic) [reply src/js/UrbanAirship.js:330]: I was following the ticket asking to return an error in the callback if the named user is not set.

Comment (BrianBatchelder) [reply src/js/UrbanAirship.js:330]: Well, that's weird. Why would we do that in frameworks if the SDKs don't do it? I feel like we should get @rlepinski to clarify.

**Comment by oristanovic** [comment]
> I think we might have to use the `--alow-warnings` flag in the pod lint step to work around the MobileCoreServices name change

It's because the SDK is not updated yet, I pushed the podspec change to make the checks pass.

**Pull request merged**